File: core/node/events/migrations/snapshot_migration_0002.go
package migrations

import (
	. "github.com/towns-protocol/towns/core/node/protocol"
	"github.com/towns-protocol/towns/core/node/shared"
)

func snapshot_migration_0002(iSnapshot *Snapshot) *Snapshot {
	switch snapshot := iSnapshot.Content.(type) {
	case *Snapshot_SpaceContent:
		for _, channel := range snapshot.SpaceContent.Channels {
			if channel.Settings == nil {
				channel.Settings = &SpacePayload_ChannelSettings{}
			}
			channelId, err := shared.StreamIdFromBytes(channel.ChannelId)
			if err != nil {
				// Note: it would be better to log this error, but we have no logging
				// context here at this time
				continue
			}
			if shared.IsDefaultChannelId(channelId) {
				channel.Settings.Autojoin = true
			}
		}
	}
	return iSnapshot
}

File: packages/sdk/src/migrations/snapshotMigration0002.ts
import { Snapshot, SpacePayload_ChannelSettings } from '@river-build/proto'
import { isDefaultChannelId, streamIdFromBytes } from '../id'

export function snapshotMigration0002(snapshot: Snapshot): Snapshot {
    switch (snapshot.content?.case) {
        case 'spaceContent': {
            snapshot.content.value.channels = snapshot.content.value.channels.map((c) => {
                if (c.settings === undefined) {
                    c.settings = new SpacePayload_ChannelSettings({
                        autojoin: false,
                        hideUserJoinLeaveEvents: false,
                    })
                }
                if (isDefaultChannelId(streamIdFromBytes(c.channelId))) {
                    c.settings.autojoin = true
                }
                return c
            })
        }
    }
    return snapshot
}
